tests/encrypt-decrypt.cpp

```diff
@@ -9,10 +9,11 @@ int main(){
     fs::path encryptedOutput("test-files/file_encrypted.txt");
     fs::path decryptedOutput("test-files/file_decrypted.txt");
 
-    encrypt(input, encryptedOutput);
-    assert(!is_data_duplicate(input, encryptedOutput));
+    encrypt(input);
+    assert(is_data_duplicate(input, encryptedOutput));
```

Comment:
This still checks different files. I think you'll need to copy the original file to a second path and compare those two files to avoid changing the test too much.

Or you could always go back to your original function implementation, and we can feed the same file in twice later on, you'll need to test that edge case too though to make sure it correctly rewrites the file since you'll have two fstreams using the same file.

Reply:
That's what I did - copied the original file into file_decrypted.txt. Or are you saying I should copy the file programmatically as part of the test?

I also put the expected output in file_encrypted.txt so I could compare it with the output after running encrypt().

Reply:
Yes I meant programmatically. But my thinking was you wouldn't be testing what you think you are.. But I suppose those files will be static. It'll just break the test if the original file is modified.

Reply:
I see the issue you're referring to but I think it's fine to leave like this for the sake of this simple exercise